go/sqltypes/proto3.go

```diff
@@ -119,6 +120,7 @@ func Proto3ToResult(qr *querypb.QueryResult) *Result {
 		Fields:              qr.Fields,
 		RowsAffected:        qr.RowsAffected,
 		InsertID:            qr.InsertId,
+		InsertIDChanged:     qr.InsertIdChanged || qr.InsertId != 0,
```

Comment:
This is just copying, we should not add a logic here. If needed, it should be done when the result is created or InsertID value is assigned the original struct.